mailpoet/lib/Entities/NewsletterEntity.php

```diff
@@ -495,7 +495,14 @@ public function getOptionValue(string $name) {
 
   public function getFilterSegmentId(): ?int {
     $optionValue = $this->getOptionValue(NewsletterOptionFieldEntity::NAME_FILTER_SEGMENT_ID);
-    return $optionValue ? (int)$optionValue : null;
+    if ($optionValue) {
+      return (int)$optionValue;
+    }
+    $parentNewsletter = $this->getParent();
+    if ($parentNewsletter instanceof NewsletterEntity && $this->getId() !== $parentNewsletter->getId()) {
```

Comment:
Would it make sense to check the newsletter type to make sure this is only applied for post notifications? Maybe it doesn't. I'm not sure.

Also, I'm curious as to why we need to check if `$this->getId() !== $parentNewsletter->getId()`? Is there a scenario where `$this->getId() === $parentNewsletter->getId()`?

Reply:
> Would it make sense to check the newsletter type to make sure this is only applied for post notifications?
My inclination is to keep this as is. I'm pretty sure parent newsletters are only used for post notifications, but if we had others or add others in the future it seems like it would make sense to defer to the parent's filter segment if the child doesn't have one of its own. I don't think it would make much of a difference right now though.

> Also, I'm curious as to why we need to check if $this->getId() !== $parentNewsletter->getId()? Is there a scenario where $this->getId() === $parentNewsletter->getId()?

Thanks for calling this out. There are a couple of tests that set the parent newsletter to itself ([example1](https://github.com/mailpoet/mailpoet/blob/360d9410f915b97c3adb053948167e2552b55fbc/mailpoet/tests/integration/Cron/Workers/SendingQueue/Tasks/NewsletterTest.php#L221-L221), [example2](https://github.com/mailpoet/mailpoet/blob/360d9410f915b97c3adb053948167e2552b55fbc/mailpoet/tests/integration/Cron/Workers/SendingQueue/Tasks/NewsletterTest.php#L240-L240)) that also preprocess the newsletter, resulting in an infinite loop. I'm almost positive we would never see this in the wild, but I would rather play it on the safe side.